### PR TITLE
Fix compact function

### PR DESCRIPTION
**Pull request by leochen1216**

### CODE REVIEW

File: lib/compass/functions/_lists.scss [hunk 1]
```diff
@@ -78,13 +78,24 @@
 }
 
 @if not(function-exists(compact)) {
-  @function compact($vars...) {
+  @function compact($args...) {
```
Comment (leochen1216) [inline lib/compass/functions/_lists.scss]: From Ruby version:
https://github.com/Compass/compass/blob/master/core/lib/compass/core/sass_extensions/functions/lists.rb#L18

File: lib/compass/functions/_lists.scss [hunk 2]
```diff
@@ -78,13 +78,24 @@
...
-        @if $var {
-            $list: append($list, $var, comma);
-        }
+
+    @if length($args) == 1 and type_of($first) == 'list' {
+      $args: $first;
+      $sep: list-separator($args);
+    }
+
+    @for $index from 1 through length($args) {
```
Comment (leochen1216) [inline lib/compass/functions/_lists.scss]: `@each $arg in $args` won't work here when the first argument is `list`. It only returns first value in the first list. 
eg: 
`((opacity .2s ease), false, false, false)`

The $arg will be `opacity` if we use `@each`

**Comment by leochen1216** [comment]
@Igosuki can you take a look at this pull request when you get a chance?

**Comment by iampuma** [comment]
Works, please merge

**Comment by xzyfer** [comment]
Thanks @leochen1216. An updated version of compact was released in v0.12.9. If you believe this to still be broken please open a PR with failing test cases.

**Comment by krivochenko** [comment]
Hi everyone!

I use version 0.12.10. And when I write: `@include  transition(color 0.5s ease-out)` I get invalid CSS:
```
transition: compact(color 0.5s ease-out, false, false, false, false, false, false, false, false, false);
```

Thanks for help.

**Comment by xzyfer** [comment]
This library is no longer supported. We recommend using bourbon.io instead.

**Pull request closed without merging**